examples/get_parent.cpp

```diff
@@ -0,0 +1,49 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#include "umpire/Allocator.hpp"
+#include "umpire/ResourceManager.hpp"
+#include "umpire/strategy/DynamicPool.hpp"
+
+void check_parent(umpire::Allocator alloc)
+{
+  umpire::strategy::AllocationStrategy* root = alloc.getAllocationStrategy();
+  while ((root->getParent() != nullptr)) {
+    root = root->getParent();
+    std::cout << root->getName() << std::endl;
+  }
+  std::cout << "(NULL)" << std::endl;
```

Comment:
I cannot remember whether resources have names or not.  If they don't then just ignore my suggestion to print the name.  But I believe this could be updated to:
```suggestion
  //
  // At this point, root will be pointing to the resource being used.
  //
  std::cout << "(root) allocator name is " << root->getName() << std::endl;
```
Of course, if I am incorrect, please just disregard.  Overall, I think this looks good and I will approve it no whether you decide to take this change or not.

Reply:
I think that's a great comment! Thanks!